lib/tools/adb-commands.js

```diff
@@ -1765,4 +1786,42 @@ methods.getTimeZone = async function getTimeZone () {
   }
 };
 
+/**
+ * Retrieves the text content of the device's clipboard.
+ * The method works for Android below and above 29.
+ * It temorarily enforces the IME setting in order to workaround
+ * security limitations if needed.
+ * This method only works if Appium Settings v. 2.15+ is installed
+ * on the device under test
+ *
+ * @returns {string} The actual content of the main clipboard
+ * or an empty string if the clipboard is empty
+ * @throws {Error} If there was a problem while getting the
+ * clipboard contant
+ */
+methods.getClipboard = async function getClipboard () {
+  log.debug('Getting the clipboard content');
+  const retrieveClipboard = async () => await this.shell([
+    'am', 'broadcast',
+    '-n', CLIPBOARD_RECEIVER,
+    '-a', CLIPBOARD_RETRIEVAL_ACTION,
+  ]);
+  let output;
+  try {
+    output = (await this.getApiLevel() >= 29)
+      ? (await this.runInImeContext(APPIUM_IME, retrieveClipboard))
+      : (await retrieveClipboard());
+  } catch (err) {
+    throw new Error(`Cannot retrieve the current clipboard content from the device. ` +
+      `Make sure the Appium Settings application is up to date. ` +
+      `Original error: ${err.message}`);
+  }
+
+  const match = /data="([^"]*)"/.exec(output);
+  if (!match) {
+    throw new Error(`Cannot parse the actual cliboard content from the command output: ${output}`);
+  }
+  return Buffer.from(_.trim(match[1]), 'base64').toString('utf8');
```

Comment:
I noticed here should be pass without decoding since the endpoint expects Base 64 encoded string.


```
# Current
[17] pry(#<AppiumLibCoreTest::Android::DeviceTest>)> c = @driver.get_clipboard
=> "\x85\xAAi\xCA\xD7\xAC\xB6)\xE0"

# Here returns `_.trim(match[1]`
[18] pry(#<AppiumLibCoreTest::Android::DeviceTest>)> @driver = @@core.start_driver
=> #<Appium::Core::Base::Driver:0x..fee76647dbb527b8e browser="unknown">
[19] pry(#<AppiumLibCoreTest::Android::DeviceTest>)> c = @driver.get_clipboard
=> "happy testing"
```

Reply:
true, will fix

thanks for checking it